lib/confluence/aggressive_removal.es6.js

```diff
@@ -43,97 +39,110 @@ foam.CLASS({
+        let postRemoval = prevLessOne.shift();
+        let currentLessOne = Array.from(current);
+        let firstCurrent = currentLessOne.shift();
 
-            // Helper: Perform SET_MINUS(removedIFaces, browserDAO's ifaces).
-            function removeFromRemoved(browserDAO) {
-              return browserDAO.interfaces.select(this.AnonymousSink.create({
-                sink: {put: function(_, iface) {
-                  if (removedIfaces.hasOwnProperty(iface.interfaceKey)) {
-                    delete removedIfaces[iface.interfaceKey];
-                  }
-                }},
-              }));
-            }
-            // Find removed APIs: APIs in browsers before date -1yr set-minus
-            // APIs in version released just before date -1yr. Also set-minus
-            // latest version of the same browser to avoid penalizing vendors
-            // that reintroduce an API after premature removal.
-            return Promise.all(promises).then(() => Promise.all([
-              removeFromRemoved.call(this, latestPrevReleaseBrowser),
-              removeFromRemoved.call(this, browser),
-            ])).then(() => {
-              // Find the interfaces that still exists in today's
-              // other browsers.
-              promises = [];
-              for (let i = 0; i < currBrowsers.length; i++) {
-                promises.push(currBrowsers[i].interfaces.select(
-                    this.AnonymousSink.create({
-                      sink: {put: function(_, iface) {
-                        if (removedIfaces.hasOwnProperty(
-                            iface.interfaceKey)) {
-                          removedIfaces[iface.interfaceKey]++;
-                        }
-                      }},
-                    })));
-              }
-              return Promise.all(promises).then(() => {
-                // Find number of APIs still exists in all today's
-                // other version of browsers.
-                let numAggressiveRemoval = Object.keys(removedIfaces).filter(
-                    (ifaceKey) => {
-                      return removedIfaces[ifaceKey] === currBrowsers.length;
-                    }).length;
-                return this.aggressiveRemovalDAO.put(
-                    this.AggressiveRemovalData.create({
-                      browserName: browser.browserName,
-                      browserOneYearAgo: latestPrevReleaseBrowser,
-                      prevReleaseBrowsers,
-                      currBrowsers,
-                      numAggressiveRemoval,
-                      date,
-                    }));
-              });
-            });
-          }));
-        }
-        return Promise.all(computeForBrowsers);
+        // Select from first current browser.
+        return firstCurrent.interfaces.select(
+            // Intersect with other current browsers to get APIs shipped by
+            // all (non-"browser" vendor) current browsers.
```

Comment:
Suggestion: Intersect with interfaces of other browsers in "current" to get a list of interfaces that exist in all of them.